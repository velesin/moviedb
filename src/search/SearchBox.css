.SearchBox {
    margin-top: 64px;
    display: flex;
}
.SearchBox__input, .SearchBox__button {
    box-sizing: border-box;
    padding: 16px 32px 16px 32px;
    font-size: 32px;
    border: none;
    outline: none;
}
.SearchBox__input {
    flex-grow: 1;
    background-color: #ffffff;
    border-top-left-radius: 8px;
    border-bottom-left-radius: 8px;
}
.SearchBox__input:focus {
    background-color: #ffffe8;
}
.SearchBox__button {
    background-color: #a0a0a0;
    border-top-right-radius: 8px;
    border-bottom-right-radius: 8px;
    cursor: pointer;
}
.SearchBox__button:hover {
    background-color: #b8b8b0;
    color: #402000;
}