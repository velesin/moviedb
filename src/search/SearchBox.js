import React, { Component } from 'react';

import './SearchBox.css';

export class SearchBox extends Component {
    state = {
        query: ''
    };    

    submitQuery = event => {
        event.preventDefault();
        this.props.onQuerySubmit(this.state.query);
    }

    updateQuery = event => {
        this.setState({ query: event.target.value });
    }

    render() {
        return (
            <form
                className="SearchBox"
                onSubmit={ this.submitQuery }
            >
                <input
                    className="SearchBox__input"
                    type="text"
                    value={ this.state.query }
                    onChange={ this.updateQuery }
                />
                <input
                    className="SearchBox__button"
                    type="submit"
                    value="Search"
                />
            </form>
        );
    }    
}