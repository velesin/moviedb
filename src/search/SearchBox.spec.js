import React from 'react';
import { shallow } from 'enzyme';

import { SearchBox } from './SearchBox';

const querySubmitSpy = jest.fn();
let searchBox;

beforeEach(() => {
    searchBox = shallow(<SearchBox onQuerySubmit={ querySubmitSpy } />);
});

it('renders search box correctly', () => {
    expect(searchBox.getElement()).toMatchSnapshot();
});

it('when user submits a query, prevents default form action and passes query up', () => {
    const QUERY = 'QUERY';
    const submitEvent = { preventDefault: jest.fn() };

    searchBox.find('input').filterWhere(node => node.prop('type') === 'text').simulate('change', { target: { value: QUERY } });
    searchBox.find('form').simulate('submit', submitEvent);

    expect(submitEvent.preventDefault.mock.calls.length).toEqual(1);
    expect(querySubmitSpy.mock.calls[0][0]).toEqual(QUERY);
});
