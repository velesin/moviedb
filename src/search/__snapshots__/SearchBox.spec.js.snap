// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search box correctly 1`] = `
<form
  className="SearchBox"
  onSubmit={[Function]}
>
  <input
    className="SearchBox__input"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    className="SearchBox__button"
    type="submit"
    value="Search"
  />
</form>
`;
